habitat/utils/profiling_utils.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+r"""Helper functions for profiling. Profilers like Nvidia Nsight can capture the
+time spent in annotated ranges.
+
+Example of range_push/range_pop usage:
+def my_function():
+    range_push("my_function")
+    for _ in range(1000)
+        range_push("my loop iteration")
+        # do stuff
+        range_pop()  # my loop iteration
+    range_pop()  # my_function
+
+Example of capturing an Nsight Systems profile:
+apt-get install nvidia-nsight
+export HABITAT_PROFILING=1
+path/to/nvidia/nsight-systems/bin/nsys profile --sample=none --trace=cuda,nvtx
+--trace-fork-before-exec=true --output=my_profile python my_program.py
+# look for my_profile.qdrep in working directory
+"""
+import os
+
+env_var = os.environ.get("HABITAT_PROFILING", "0")
+enable_profiling = env_var != "0"
+if enable_profiling:
+    print("HABITAT_PROFILING={}".format(env_var))
+    print("profiling_utils.py range_push/range_pop annotation is enabled")
+    from torch.cuda import nvtx
+
+
+def range_push(msg: str):
```

Comment:
@eundersander The proposed usage is actually not ideal. I would much more recommend creating a context object and using `__enter__` and `__exit__`. That would be much more pythonic and much more resilient to errors.

Reply:
Hey, a context object is a great suggestion. I will add it here, built on top of range_push/range_pop.

I personally find context objects annoying to use for this kind of thing, because they require you to indent huge blocks of code just to add them, creating diff spam and making rapid iteration difficult. But to each their own; it's user preference. I agree they are less error-prone.

Reply:
@eundersander You are right, the best thing here would be a function decorator!

Reply:
@eundersander Extend https://docs.python.org/3/library/contextlib.html#contextlib.ContextDecorator to use it as both!